src/cursor/abstract_cursor.ts

```diff
@@ -284,11 +286,34 @@ export abstract class AbstractCursor<
   }
 
   [Symbol.asyncIterator](): AsyncIterator<TSchema, void> {
+    async function* nativeAsyncIterator(this: AbstractCursor<TSchema>) {
+      if (this.closed) {
+        return;
+      }
+
+      while (true) {
+        const document = await this.next();
+
+        if (document == null) {
```

Comment:
I was looking at refactoring `next` on Friday, and I think it's slightly more complicated than this.  This forbids any values of `null` from being emitted from the cursor, so something like 

```typescript
for await (const document of cursor.map(() => null) {

}
```

will not actually iterate the cursor.

I think we need a sort of Result-like abstraction to return from `next` (I'm looking at it now actually) because `next` needs to differentiate between the cursor being closed (currently `next` returns null in this scenario) and handling a `null` document.  If `next` instead returned something like `{ value: T | null } | null`, we could rely on `null` indicating that the cursor was exhausted.